### ISSUE TITLE
Clean up the writing of recording messages.

### ISSUE THREAD

**Issue by kowalej** [labstreaminglayer/App-LabRecorder#12]
Since some messages such as "Offsets thread is finished.", "Received header for stream <xyz>", etc, are being outputted from multiple threads, they show up looking a mess inside a console window, especially when one has multiple LSL streams running. Perhaps you can lock a new "output mutex" when doing calls to cout and cerr to make sure the messages look nice.

**Comment by tstenner** [comment]
So far I haven't had that issue personally, but I mostly have less than 10 streams and maybe my libstdc++ is less prone to it than others.

My suggestion: create a logging function (`void log(const std::string& msg)`) that locks a mutex internally. That way we can also enable callbacks to a logging function later on more easily.
Is this something you'd be comfortable preparing a PR for?

**Comment by kowalej** [comment]
Yep that's exactly what I was thinking. I can do a PR for it pretty soon. For reference this messy message issue was seen on multiple different Windows 10 machines in both CMD and PowerShell outputs.
